kore/src/Kore/Internal/SideCondition/SideCondition.hs

```diff
@@ -34,7 +36,9 @@ import Type.Reflection
     )
 
 data Representation where
-    Representation :: (Ord a, Pretty a) => !(TypeRep a) -> !(Hashed a) -> Representation
+    Representation
+        :: (Ord a, Pretty a)
+        => ~(TypeRep a) -> ~(Hashed a) -> Representation
```

Comment:
It should be completely fine to have the `TypeRep` be strict (it's just a pointer to a single-constructor type with no arguments).

```suggestion
        => !(TypeRep a) -> ~(Hashed a) -> Representation
```